sklearn/feature_extraction/tests/test_text.py

```diff
@@ -995,3 +998,28 @@ def test_vectorizer_string_object_as_input():
             ValueError, message, vec.fit, "hello world!")
         assert_raise_message(
             ValueError, message, vec.transform, "hello world!")
+
+
+@pytest.mark.parametrize("dtype", [
+    np.float32,
```

Comment:
you need to parametrize the dtype of X and the dtype of the tfidfvectorizer to try all combination.

I would suspect that right now:

* X -> np.float32, Vectorizer -> np.float64
* X -> np.float64, Vectorizer -> np.float32

will failed.

Reply:
I've added separate cases for Vectorizer and Transformer. I hope you mean the same thing?